### PR TITLE
42) Fix for terrain memory leaks

### PR DESCRIPTION
**Pull request by tkgdhughes**
### Description 

Fixed memory leaks in terrain_node_compile.cpp and terrain_sector.h

### CODE REVIEW

**Comment by ranilianaws** [comment]
Hey @tkgdhughes ,

Thank you for the contribution. We will evaluate the code and get back to you soon!

**Comment by tkgdhughes** [comment]
Hey @ranilianaws, 

I'll check the file again, I think you have a good point!

File: dev/Code/CryEngine/Cry3DEngine/terrain_node_compile.cpp [hunk 1]
```diff
@@ -86,6 +87,7 @@ int CTerrainNode::Load_T(T& f, int& nDataSize, EEndian eEndian, bool bSectorPale
         ITerrain::SurfaceWeight* weights = new ITerrain::SurfaceWeight[(ChunkSizeSqr + 7) & (~0x7)];
         if (!CTerrain::LoadDataFromFile(weights, ChunkSizeSqr, f, nDataSize, eEndian))
         {
+            delete[] weights;
```
[the remark this answers is not in the document]
Comment (tkgdhughes) [reply dev/Code/CryEngine/Cry3DEngine/terrain_node_compile.cpp]: Good spot, making the change and pushing it this morning.

File: dev/Code/CryEngine/Cry3DEngine/terrain_node_compile.cpp [hunk 2]
```diff
@@ -118,6 +120,7 @@ int CTerrainNode::Load_T(T& f, int& nDataSize, EEndian eEndian, bool bSectorPale
 
             if (!CTerrain::LoadDataFromFile(pTypes, chunk.nSurfaceTypesNum, f, nDataSize, eEndian))
             {
+                delete[] pTypes;
```
[the remark this answers is not in the document]
Comment (tkgdhughes) [reply dev/Code/CryEngine/Cry3DEngine/terrain_node_compile.cpp]: This is fine it appears, ownership is transferred above through `m_SurfaceTile.AssignMaps(chunk.nSize, heightmap, weights);`
There are a lot of raw pointers sprinkled throughout this file and we should deprecate them all and replace with something smarter, but it was a bigger change than we wanted to make to directly address the leaks.

**Comment by ranilianaws** [comment]
Hey @tkgdhughes ,

Thank you for the contribution. We have accepted the change into a future release of LY.

**Pull request closed without merging**